memory/src/main/scala/filodb.memory/data/OffheapLFSortedIDMap.scala

```diff
@@ -300,19 +302,20 @@ class OffheapLFSortedIDMapReader(memFactory: MemFactory, holderKlass: Class[_ <:
   def acquireExclusive(inst: MapHolder): Unit = {
     // Spin-lock implementation. Because the owner of the shared lock might be blocked by this
     // thread as it waits for an exclusive lock, deadlock is possible. To mitigate this problem,
-    // timeout and retry, allowing shared lock waiters to make progress.
+    // timeout and retry, allowing shared lock waiters to make progress. The timeout doubles
+    // for each retry, up to a limit, but the retries continue indefinitely.
 
-    var timeoutNanos = 1000000 // 1 millisecond
+    var timeoutNanos = 1.millisecond.toNanos
     var warned = false
 
     while (true) {
       if (tryAcquireExclusive(inst, timeoutNanos)) {
         return
       }
 
-      timeoutNanos = Math.min(timeoutNanos << 1, 1000000000) // limit timeout to 1 second
+      timeoutNanos = Math.min(timeoutNanos << 1, 1.second.toNanos) // limit timeout
```

Comment:
What I meant was to move this `1.second.toNanos` into a global val so that the cost of doing this conversion is not in every call.....

Reply:
Ok.